### DISCUSSION TITLE
Tools for Logging and Monitoring, backup/Restore Oracle 19c Single instance on Kubernetes

### DISCUSSION THREAD

**Discussion by Champak-Netcracker** [oracle/docker-images#2039] [Q&A]
Hello @gvenzl , team -

Monitoring and Logging:
What tools/software that are either Open Source or Oracle provided tools that are supported by Oracle[subject to Netcracker Licensing Model] that can be used:
A) Monitoring solution: export metrics to dashboard reports, Grafana, etc
B) Logging solution: export critical log messages for analytics to tools like Log Stash, Kibana etc
C) Can we use Prometheus/Grafana/ELK as a Logging and Monitoring solution, is this supported by Oracle database 19c running on Kubernetes, if yes, pls also share the documentation link?
D) Based on the tools, can we raise any support SR with Oracle?
E) Should I also raise a support SR and post these queries for parallel record and redirect them to you folks? 

**Backup/Restore:**
What tools/software that are either Open Source or Oracle provided tools that are supported by Oracle[subject to Netcracker Licensing Model] that can be used:
A) to Initiate Backup/restore operations
B) Pls also share the documentation link?
C) Any built in APIs to store the backup on Object Storage services on Cloud platform likes AWS, GCP, Azure, Oracle Cloud, OpenShift Container platform/OpenStack
D) Based on the tools, can we raise any support SR with Oracle?
E) Should I also raise a support SR and post these queries for parallel record and redirect them to you folks? 

Thanks/Champak

**Comment by Champak-Netcracker** [comment]
@gvenzl /team- Any update on this please?

**Comment by gvenzl** [comment]
Hi @yunus-qureshi, @abhisbyk and team, can you please look at the questions and provide the corresponding answers and/or best practices?

**Comment by Champak-Netcracker** [comment]
Hello @yunus-qureshi, @abhisbyk - Please let me know your responses, its been a while since I have had any inputs? please also answer specific to the points mentioned above?

**Reply by abhisbyk** [reply]
Hi @Champak-Netcracker
We are internally discussing on monitoring, logging and backup/restore solutions for SingleInstance Database on Kubernetes. We will revert back soon. Thanks.

**Reply by Champak-Netcracker** [reply]
Hi @abhisbyk - Please let me know when I can expect some responses?

**Reply by Champak-Netcracker** [reply]
@abhisbyk , I haven't see any update, please let me know?

**Reply by abhisbyk** [reply]
Hi @Champak-Netcracker,
Sorry for the delay in response. We are currently developing solutions in monitoring/logging, backup/restore directions and are in approval process. It would take around 6 to 8 weeks before this topic gains clarity.
Thanks

**Reply by Champak-Netcracker** [reply]
Hi @abhisbyk , thanks, please keep me posted once it is available for customers. I will keep this thread open until then. 

/Champak

**Reply by Champak-Netcracker** [reply]
Hi @gvenzl /@abhisbyk - Do you have any update on this?

/Champak

**Reply by Champak-Netcracker** [reply]
Hi @gvenzl /@abhisbyk - Do you have any update on this?

/Champak

**Reply by abhisbyk** [reply]
Hi @Champak-Netcracker !!

Oracle Database Operator for kubernetes (a.k.a. OraOperator) is released now. Currently, it supports database provisioning, patching and cloning for Oracle Database 21c. The repo link is [here](https://github.com/oracle/oracle-database-operator).
The OraOperator is going to be the solution for monitoring, backup/restore etc for Oracle Database. In next drops of OraOperator we are planning to release Oracle REST Data Services i.e. ORDS (can be used for monitoring as well), APEX, backup/restore support. It requires some more weeks for these solutions to be out.

Thanks
Abhishek

**Reply by Champak-Netcracker** [reply]
@abhisbyk , great news, many thanks for sharing this information, we look forward to hear from you once OraOperator supports Logging and Monitoring, backup/Restore, please keep us posted.

**Reply by abhisbyk** [reply]
Yes, sure. I will keep you posted.
Additionally, you can refer to this [blog](https://bit.ly/3jDHCfU) for a very good introduction of the OraOperator.

**Comment by Champak-Netcracker** [comment]
@gvenzl or @abhisbyk - Trust you are doing great, in relation to the OraOperator, can you please let us know the following:
Does it support provisioning Oracle 18c/19c?
Does it support patching Grid/RAC/DB?

**Reply by abhisbyk** [reply]
@Champak-Netcracker OraOperator supports Oracle Single Instance Database 19c onwards (currently 19c and 21c). And, it also supports patching for Single Instance DB. Thanks

**Comment by Champak-Netcracker** [comment]
@abhisbyk , so from the above, I understand, the following:
- OraOperator does not support provisioning 19c/21c RAC DB?
- Based on the above point, OraOperator does not support Grid, RAC, ASM patching? Please confirm?
Any ETA when this will be available? Please let me know?

**Reply by abhisbyk** [reply]
@Champak-Netcracker No, RAC DB is not supported for now. Neither patching is supported for Grid, RAC and ASM. Patching is only supported for SIDB. Full features of the OraOperator are [here](https://github.com/oracle/oracle-database-operator#features-summary). 
At this point of time we can not specify any ETA for these features. Internal discussions are going on.
Thanks

**Comment by Champak-Netcracker** [comment]
@abhisbyk OK, Thanks, can you please clarify this, 
1) From a design point of view Kubernetes provides pod level HA and scalability, and if underlying Infra layer is HA, and single instance database is configured  with PV, do we need RAC? or am I missing any cache part? with pod replicas, can it handle? Please help with my understanding?
2) Can you please let me know if AWS cloud has any Oracle RAC DB offerings?
3) Does Oracle support RAC DB on other Cloud providers?

**Reply by Champak-Netcracker** [reply]
@abhisbyk -Appreciate your responses.

Thx/Champak

**Reply by Champak-Netcracker** [reply]
@abhisbyk- please help clarify above?

**Reply by abhisbyk** [reply]
@Champak-Netcracker As far as I know, Oracle has not certified RAC DB on any other cloud platforms. I will get back after the discussion with the RAC team to bring you more clarity on your questions.

**Comment by abhisbyk** [comment]
@Champak-Netcracker Application continuity is not available with SIDB on Kubernetes running in multiple replica mode. In other words, if one pod goes down, the other replica pod takes it's place but the database connection is broken for that time. So, with pod replicas, HA is not seamless. But, Oracle RAC provides seamless connectivity and performance.

@psaini79 Please help here to gain more clarity for Oracle RAC based solutions.

**Reply by Champak-Netcracker** [reply]
@abhisbyk , thanks, much appreciated, I will wait for @psaini79 to add more clarity.

/Champak

**Reply by Champak-Netcracker** [reply]
@psaini79 - Gentle reminder, please add more clarity? Thanks

**Reply by Champak-Netcracker** [reply]
@gvenzl - Can you please help us here to seek clarity on the above RAC explanation? We are also in parallel try to check if we have some single instance database that we can modernize for our lower environments, but at the same time, we need the RAC option with the Operator charts, eventually helping us with an automation journey for some of our Oracle database customers? Appreciate your support here? My regards to all your team! Stay safe in these times. 

Thanks/Champak

**Reply by Champak-Netcracker** [reply]
@gvenzl or @abhisbyk  - I would also like to understand, if we have any github documentation to automate GRID/RAC DB 19.XX EE On Enterprise Linux/ VM based for On Premise customers, we are trying to help some of our Oracle database customers to achieve this? appreciate your insights/help? Also, some of our customers{holds good for the above request} choose to remain On Premise for security reasons"

**Reply by Champak-Netcracker** [reply]
@abhisbyk and @gvenzl , we found these blogs to help with 19c Grid/RAC DB install/patching for on premise based deployments, please let us know if these will be useful for my above query, I believe some of these experts are part of Oracle development and hence the clarification? Thanks

**Reply by Champak-Netcracker** [reply]
https://blog.iarsov.com/oracle/automating-oracle-patching-with-ansible/
https://github.com/iarsov/ansible-orapatch
https://github.com/klewan/ansible-role-oracle-manage-patches
https://github.com/CruGlobal/ansible-oracle-psu
https://github.com/chef1729/dbpatching

https://medium.com/oracledevs/devops-series-automate-oracle-19c-grid-infrastructure-installations-with-ansible-github-bc491bdd34ce
https://github.com/oravirt/ansible-oracle
https://medium.com/oracledevs/devops-series-automate-of-oracle-19c-rac-release-update-from-19-3-to-19-4-using-ansible-part-iv-475c1da7cf0e

**Comment by Champak-Netcracker** [comment]
@abhisbyk - In the previous update from you "The OraOperator is going to be the solution for monitoring, backup/restore etc for Oracle Database", can you please let us know if the logging/monitoring, Backup/restore solutions are out?

**Comment by Champak-Netcracker** [comment]
@abhisbyk - In the previous update from you "The OraOperator is going to be the solution for monitoring, backup/restore etc for Oracle Database", can you please let us know if the logging/monitoring, Backup/restore solutions are out?

**Comment by Champak-Netcracker** [comment]
@gvenzl  or @abhisbyk - Any update on the above please?

**Comment by abhisbyk** [comment]
@Champak-Netcracker The next drop of OraOperator is going to be out in next couple of months. It will contain the monitoring solution for DB. The third drop will contain backup solution, timeline of which is not fixed yet. Thanks !

**Comment by Champak-Netcracker** [comment]
OK thanks @abhisbyk

**Comment by Champak-Netcracker** [comment]
hi @gvenzl - Trust you are doing well, based on the attached public content/link: please clarify if Oracle is now certifying RAC on other Cloud platforms[IaaS/Kubernetes], containers, On premise[VMs, BMs]? if yes, do you have any MOS IDs for reference?

https://www.slideshare.net/MarkusMichalewicz/oracle-rac-virtualized-in-vms-in-containers-onpremises-and-in-the-cloud

Have a nice day!

Cheers/Champak

**Reply by gvenzl** [reply]
Hi @psaini79, 

Can you please help clarify the situation for Oracle RAC certification in a containerized environment?

Thanks,

**Reply by Champak-Netcracker** [reply]
@psaini79 - Any update on this? appreciate it?

**Comment by Champak-Netcracker** [comment]
@gvenzl and @psaini79 - Appreciate if I can receive some confirmation around this:

please clarify if Oracle is now certifying RAC on other Cloud platforms[IaaS/Kubernetes], containers, On premise[VMs, BMs]? if yes, do you have any MOS IDs for reference?

https://www.slideshare.net/MarkusMichalewicz/oracle-rac-virtualized-in-vms-in-containers-onpremises-and-in-the-cloud

**Comment by Champak-Netcracker** [comment]
@gvenzl - Hi GM, its been 3 weeks since I have been requesting updates on this, can you please help here?

**Comment by Champak-Netcracker** [comment]
@psaini79  AND @gvenzl - Do you have any updates, any reason why there is a delay?

**Comment by gvenzl** [comment]
Hi @Champak-Netcracker,

I have just spoken with my colleague and author of the slides you have referenced above and he requested that you reach out directly to him via email, please. His email address is <first_name>.<last_name>@oracle.com

Thanks,

**Reply by Champak-Netcracker** [reply]
@gvenzl - Appreciate your response, let me internally discuss with our leadership and come back. Thanks/Champak

**Comment by Champak-Netcracker** [comment]
Hello @gvenzl - We need clarity on a few queries here before we reach out to Markus, please can you help answer the below:
we have two questions here : based on this MOS note: [Support Policy for Oracle Database Container Images on Kubernetes Clusters (Doc ID 2789626.1)], A) Oracle RAC container images are not supported on non-Oracle public cloud environments, ex: AKS, EKS, Redhat OpenShift, etc. Is this understanding correct? B) Is RAC docker images supported on On-premise[VM/BM], core K8s?

**Comment by gvenzl** [comment]
Hi @Champak-Netcracker,

I'm very sorry but I cannot provide you with any of these answers as Markus is the person responsible for the RAC on Docker area.
Please reach out to him directly with these questions.

Thanks,
